article/_posts/2020-11-19-AlphaMuGoZero.md

```diff
@@ -0,0 +1,181 @@
+
+$$v_\theta (s) = \mathbb{E}[z_t | s_t=s, a_{t...T} \sim p_\rho]$$
+
+with $$z$$ being the outcome of the game, $$\pm 1$$ depending on the player. $$v_\theta$$ was trained on a collection of 30 million game states, each from a different game played by versions of $$p_\rho$$ so as to provide decorrelated labeled data thus reducing overfitting.
+
+![](/article/images/alpha/fig2.jpg)
+> Silver, D., Huang, A., Maddison, C. et al. Mastering the game of Go with deep neural networks and tree search. Nature 529, 484–489 (2016). https://doi.org/10.1038/nature16961
+
+Once training is over, MCTS is used to play the game. At each node, the action $$a$$ is selected according to
+
+$$a_t = \argmax_a (Q(s_t,a) + u(s_t,a))$$
+
+with 
+
+$$u \propto \frac{P(s,a)}{1 + N(s,a)}$$
+
+$$P(s,a) = p_\sigma(s,a)$$
+
+$$N(s,a) =  \sum_{i=1}^N 1(s,a,i)$$
+
+$$Q(s,a) = \frac{1}{N(s,a)}\sum_{i=1}^n 1(s,a,i)V(s_L^i)$$
+
+$$V(s_L) = (1-\lambda)v_\theta (s_L) + \lambda z_L$$
+
+$$p_\pi$$ is used during rollouts due to its smaller size and therefore faster inference.
+
+![](/article/images/alpha/fig3.jpg)
+> Silver, D., Huang, A., Maddison, C. et al. Mastering the game of Go with deep neural networks and tree search. Nature 529, 484–489 (2016). https://doi.org/10.1038/nature16961
+
+Above is the ELO rating for AlphaGO and European Go champion Fan Hui, as well as other Go programs and different versions of AlphaGo. AlphaGo went on to [beat Lee Sedol 4-1 afterward](https://www.youtube.com/watch?v=WXuK6gekU1Y).
+
+## AlphaGo Zero
+
+In a subsequent paper, Deepmind presented AlphaGo Zero, which achieves higher performance than AlphaGo using fewer priors.
+
+![](/article/images/alpha/fig4.jpg)
+> Silver, D., Schrittwieser, J., Simonyan, K. et al. Mastering the game of Go without human knowledge. Nature 550, 354–359 (2017). https://doi.org/10.1038/nature24270
+
+A couple of differences are to note:
+- No supervised learning is used to initialize policies
+- A single policy is used
+- A single network is used for both the policy and the value function
+- The network is fed raw board configurations instead of handcrafted features
+
+This new version of the algorithm takes the form
+
+$$p, v = f_\theta(s)$$
+
+, meaning that the network outputs both action probabilities $$p$$ and value estimation $$v$$ for states $$s$$. During training, at each move, the network performs MCTS simulations following
+
+$$a_t = \argmax_a U(s,a)$$
+
+with
+
+$$U(s,a) = Q(s,a) + c \cdot p(s,a) \cdot \frac{\sqrt{\sum_b N(s,b)}}{1+N(s,a)}$$
+
+$$Q(s,a) = \frac{1}{N(s,a)}\sum_{s'|s,a\rightarrow s'}v(s')\; \text{(an average of the values of the nodes from s' to s)}$$ 
+
+and $$c$$ a parameter promoting exploration. However, instead of performing rollouts, the network uses $$v(s_L)$$ to predict the outcome. MCTS' execution gives the tuple $$(s, \pi, p, \_)$$ where $$s$$ is the game's state, $$\pi$$ corresponds to the normalized state/actions visitations
```

Comment:
Not sure whether the term "visitation" is appropriate in this context.

Reply:
Reworded in 137ef52